airflow-core/src/airflow/ui/src/pages/DagsList/DagsList.tsx

```diff
@@ -203,16 +203,26 @@ export const DagsList = () => {
   const { setTableURLState, tableURLState } = useTableURLState();
 
   const { pagination, sorting } = tableURLState;
+  const [savedSearchPattern, setSavedSearchPattern] = useLocalStorage("dags_search_temp", "");
   const [dagDisplayNamePattern, setDagDisplayNamePattern] = useState(
-    searchParams.get(NAME_PATTERN) ?? undefined,
+    searchParams.get(NAME_PATTERN) ?? savedSearchPattern,
   );
 
+  useEffect(() => {
+    const handleBeforeUnload = () => setSavedSearchPattern("");
+
+    globalThis.addEventListener("beforeunload", handleBeforeUnload);
+
+    return () => globalThis.removeEventListener("beforeunload", handleBeforeUnload);
+  }, [setSavedSearchPattern]);
+
```

Comment:
Can we make the same without `useEffect()`?

Reply:
I think I can use sessionStorage instead of localStorage. 
That way, I might be able to avoid using useEffect. I’ll try it ;)